app/src/lib/error-with-metadata.ts

```diff
@@ -15,6 +16,9 @@ export interface IErrorMetadata {
 
   /** Additional context that specific actions can provide fields for */
   readonly gitContext?: GitErrorContext
+
+  /** The branch intended to be checked out */
+  readonly checkoutBranch?: Branch
```

Comment:
I think this should be in a new type of `GitErrorContext` instead of added directly in `IErrorMetadata` since this field is very specific to bringing changes during switching branches.

Reply:
Addressed in 7717a2b.